app/containers/MessageComposer/components/ComposerInput.tsx

```diff
@@ -352,6 +355,27 @@ export const ComposerInput = memo(
 			dispatch(userTyping(rid, isTyping));
 		};
 
+		const navigation = useNavigation();
+
+		useEffect(() => {
+			if (isIOS) {
+				const startListener = navigation.addListener('transitionStart' as any, e => {
+					if (e?.data?.closing) {
+						iOSBackSwipe.current = true;
+					}
+				});
+
+				const newListener = navigation.addListener('blur', () => {
+					Keyboard.dismiss();
+				});
+
+				return () => {
+					startListener();
+					newListener();
+				};
+			}
+		}, [navigation]);
```

Comment:
Of course we don't want this workaround, but if we need it, create a separate hook to isolate this workaround away from ComposerInput